app/packages/app/src/useEvents/utils.ts

```diff
@@ -24,4 +24,32 @@ export const processState = (
     toCamelCase(state.selected_labels) as State.SelectedLabel[]
   );
   state.spaces && setter("sessionSpaces", state.spaces);
+
+  const visibility = state.field_visibility_stage;
+  const visibilityStage = visibility
+    ? {
+        _cls: visibility.cls,
+        kwargs: [
+          ["field_names", visibility.field_names],
+          ["_allow_missing", true],
+        ],
+      }
+    : null;
+  const stageAtom = visibility
+    ? {
+        cls: visibility.cls,
+        kwargs: {
+          field_names: visibility.field_names,
+          allow_missing: true,
+        },
+      }
+    : undefined;
+
+  stageAtom && setter("fieldVisibilityStage", stageAtom);
+  return env().VITE_NO_STATE
+    ? { view: [], extendedStages: [] }
+    : {
+        view: state.view || [],
+        extendedStages: visibilityStage ? [visibilityStage as State.Stage] : [],
```

Comment:
Can we use a utility function to append `extendedStages`, as we are adding other type of extendedStages.

Reply:
Yeah, I will think about how to clean this up. It's a good idea

Reply:
I'm going to try improving this in #3643 once this is merged. Some work is already done there, definitely agree there is  an opportunity to simplify and add reuse